src/test/functions.php

```diff
@@ -2,19 +2,65 @@
 
 require '../functions.php';
 
-use PHPUnit\Framework\TestCase; // magic to get PHPUnit testing to work (lookup)
+use PHPUnit\Framework\TestCase;
 
 class Functions extends TestCase
 {
-	public function test_Success_Display_Cosmic_Event() {
-		$input = [
+	public function test_success_display_cosmic_event() {
+		$db_row_input = [
 			'id' => 1,
 			'name' => 'Big Bang',
 			'caused_by' => 'Unknown',
 			'creates_effect' => 'Fusion'];
+		$event_elements = ['Unit', 'Testing', 'Is', 'Super', 'Fun'];
 		$expected_output = '<div class="cosmic_event"><h2>Big Bang</h2><h3>Caused by: Unknown</h3><h3>Creates: '
-			. 'Fusion</h3><h3>Produces: </h3></div>';
-		$actual_output = display_Cosmic_Event($input);
+			. 'Fusion</h3><h3>Produces:</h3><p>Unit, Testing, Is, Super, Fun, </p></div>';
+		$actual_output = display_cosmic_event($db_row_input, $event_elements);
+		$this->assertEquals($expected_output, $actual_output);
+	}
+
+	public function test_failure_display_cosmic_events() {
+		$db_row_input = [
+			'id' => 1,
+			'name' => 'Big Bang',
+			'caused_by' => 'Unknown',
+			'will_wonker' => 'Fusion'];
+		$event_elements = ['Unit', 'Testing', 'Is', 'Super', 'Fun'];
+		$expected_output = 'FAIL';
+		$actual_output = display_cosmic_event($db_row_input, $event_elements);
 		$this->assertEquals($expected_output, $actual_output);
 	}
+
+	public function test_malformed_display_cosmic_event() {
+		$db_row_input = 1;
+		$event_elements = [1, 2, 5, 7];
+		$this->expectException(TypeError::class);
+		$output = display_cosmic_event($db_row_input, $event_elements);
+	}
+
+	public function test_success_format_event_elements() {
+		$input = [
+			['name' => 'Hydrogen'],
+			['name' => 'Helium'],
+			['name' => 'Lithium']];
```

Comment:
the new lines are a bit off - maybe either put [[ at the start or put a line break before the final ]